core/Updates/4.0.0-b1.php

```diff
@@ -110,6 +113,34 @@ public function getMigrations(Updater $updater)
         // remove old options
         $migrations[] = $this->migration->db->sql('DELETE FROM `' . Common::prefixTable('option') . '` WHERE option_name IN ("geoip.updater_period", "geoip.loc_db_url", "geoip.isp_db_url", "geoip.org_db_url")');
 
+        // replace days to ... dimensions w/ seconds dimensions
+        foreach (['log_visit', 'log_conversion'] as $table) {
+            $migrations[] = $this->migration->db->addColumn($table, 'visitor_seconds_since_first', VisitorSecondsSinceFirst::COLUMN_TYPE);
+            $migrations[] = $this->migration->db->addColumn($table, 'visitor_seconds_since_order', VisitorSecondsSinceFirst::COLUMN_TYPE);
+        }
+
+        $migrations[] = $this->migration->db->addColumn('log_visit', 'visitor_seconds_since_last', VisitorSecondsSinceLast::COLUMN_TYPE);
+
+        $tableMetadata = new TableMetadata();
+        $logVisitColumns = $tableMetadata->getColumns(Common::prefixTable('log_visit'));
+        $hasDaysColumnInVisit = in_array('visitor_days_since_first', $logVisitColumns);
+
+        $logConvColumns = $tableMetadata->getColumns(Common::prefixTable('log_conversion'));
+        $hasDaysColumnInConv = in_array('visitor_days_since_first', $logConvColumns);
+
+        if ($hasDaysColumnInVisit) {
+            $migrations[] = $this->migration->db->sql("UPDATE " . Common::prefixTable('log_visit')
+                . " SET visitor_seconds_since_first = visitor_days_since_first * 86400, 
+                    visitor_seconds_since_order = visitor_days_since_order * 86400,
+                    visitor_seconds_since_last = visitor_days_since_last * 86400");
+        }
+
+        if ($hasDaysColumnInConv) {
+            $migrations[] = $this->migration->db->sql("UPDATE " . Common::prefixTable('log_conversion')
+                . " SET visitor_seconds_since_first = visitor_days_since_first * 86400, 
+                    visitor_seconds_since_order = visitor_days_since_order * 86400");
+        }
```

Comment:
do I see this right we keep the old columns `visitor_days_since_first` etc?

Reply:
to my understanding deleted dimensions aren't actually removed, so this is ok? otherwise I can delete the column as well.

Reply:
I reckon we delete the columns as there wouldn't be any use for them anymore. Unless I'm not aware of something @mattab maybe knows more.

In https://github.com/matomo-org/matomo/wiki/Matomo-4.0.0-release-notes#downgrading-to-matomo-3 we could add some queries to add back the columns again if needed and we could even migrate the old column values to what they were before basically.